osf/models/user.py

```diff
@@ -777,6 +779,24 @@ def merge_user(self, user):
 
         user.save()
 
+    def _merge_users_preprints(self, user):
+        from osf.models.preprint import Preprint, PreprintContributor
+
+        preprints = Preprint.objects.filter(creator=user)
+        for preprint in preprints:
+            preprint.creator = self
+            preprint.save()
+
+        preprint_contributors = PreprintContributor.objects.filter(user=user)
+
+        # If both users are already contribs we will violate unique constraint so just remove old contrib.
+        for preprint_contributor in preprint_contributors:
+            if self in preprint_contributor.preprint.contributors.all():
+                preprint_contributor.delete()
```

Comment:
[We need to do the same thing for preprints that we do when merging projects on users. ](https://github.com/CenterForOpenScience/osf.io/blob/develop/osf/models/user.py#L702)Before you delete the contributor, you need to check permissions and bibliographic information. If both users belong to the preprint, then the higher permission of either user needs to be retained.  If one user is visible, and the other is not, the merged user might need to be updated to become visible. 

For example, if John and John1 both belong to a preprint and John has `write` perms, and John2 has `admin` perms, if we merge John <-- John2, then John needs to be given admin perms in the merge, an upgrade.   If John is not visible, and John2 is visible, we need to make John visible. 

Update `self contributor`'s permissions and bibliographic information before deleting the `user contribtor`, seeing project link above for an example.